GoogleForCreators/web-stories-wp
# Page templates: adding and displaying custom templates

## Context

This PR adds the part for adding and displaying the custom templates.

Note that it does not include displaying the Save button for the default templates yet.


## Summary



## Relevant Technical Choices

- Uses just a normal grid for the list of page templates instead of the virtual list as for the default templates. The simple grid just seems to be simpler and more suitable use case here, additionally, the templates are loaded via API instead of having them all together at once, meaning that we'll need to implement pagination, as in the case of Media. Edit: This will be rethought in a separate issue for performance reasons, potentially to use virtual list here, too.


## To-do



## User-facing changes



## Testing Instructions
- Turn on the Custom page templates experiment.
- Open the Page templates section in the Library and switch to "Saved Templates"
- Try clicking "Save" with an empty Page
- Verify a template is not added and a snackbar message is displayed to inform about not being able to add empty Page as a template.
- Add some elements to the Page, add some animation, too.
- Try saving the template now.
- Verify the template was added. Verify that hovering on the template will play the animation.


### QA
See the testing instructions above





This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

### Does this PR change what data or activity we track or use?
This will now track using the saved templates, too.


## Checklist




---



Fixes #7060 
Fixes #7064
